modules/core/src/main/java/org/apache/ignite/internal/processors/query/schema/SchemaOperationWorker.java

```diff
@@ -177,8 +178,17 @@ public boolean cacheRegistered() {
      * Cancel operation.
      */
     @Override public void cancel() {
-        if (cancelToken.cancel())
+        if (cancelToken.cancel()) {
+            try {
+                fut.get();
```

Comment:
Do you have any guarantees that this future will finish eventually? I just afraid that this line can lead to the hung of the node.